src/code/m_npc_walk.c

```diff
@@ -0,0 +1,659 @@
+#include "m_npc_walk.h"
+#include "m_common_data.h"
+#include "m_npc.h"
+#include "m_random_field.h"
+#include "sys_math.h"
+#include "m_field_info.h"
+#include "macros.h"
+#include "attributes.h"
+
+#define mNpcW_TIME_2_SEC(hour, min, sec) ((hour)*3600 + (min)*60 + (sec))
+#define mNpcW_MAKE_SCHEDULE_TABLE(goal_data) \
+    { goal_data, ARRAY_COUNT(goal_data) }
+
+s32 l_goal_block[NPCW_GOAL_BLOCK_NUM][2];
+s32 l_arrive_stay_count[ANIMAL_NUM_MAX];
+
+u8 l_girl_time_12[] = { NPCW_GOAL_SHRINE, NPCW_GOAL_HOME, NPCW_GOAL_ALONE };
+u8 l_girl_time_18_30[] = { NPCW_GOAL_SHRINE, NPCW_GOAL_HOME };
+
+mNpcW_GoalData_c l_girl_goal_data[] = { { NULL, 0, mNpcW_TIME_2_SEC(6, 0, 0) },
+                                        { l_girl_time_12, ARRAY_COUNT(l_girl_time_12), mNpcW_TIME_2_SEC(12, 0, 0) },
+                                        { NULL, 0, mNpcW_TIME_2_SEC(13, 0, 0) },
+                                        { l_girl_time_18_30, ARRAY_COUNT(l_girl_time_18_30),
+                                          mNpcW_TIME_2_SEC(18, 30, 0) },
+                                        { NULL, 0, mNpcW_TIME_2_SEC(24, 0, 0) } };
```

Comment:
Could you add a trailing comma to this array and run formater again? It usually looks a bit better this way
```suggestion
mNpcW_GoalData_c l_girl_goal_data[] = { { NULL, 0, mNpcW_TIME_2_SEC(6, 0, 0) },
                                        { l_girl_time_12, ARRAY_COUNT(l_girl_time_12), mNpcW_TIME_2_SEC(12, 0, 0) },
                                        { NULL, 0, mNpcW_TIME_2_SEC(13, 0, 0) },
                                        { l_girl_time_18_30, ARRAY_COUNT(l_girl_time_18_30),
                                          mNpcW_TIME_2_SEC(18, 30, 0) },
                                        { NULL, 0, mNpcW_TIME_2_SEC(24, 0, 0) }, };
```